### PR TITLE
multiple solver support in Matlab and sfunction compilation on Windows

### PR DESCRIPTION
**Pull request by asparc**

### CODE REVIEW

File: examples/acados_matlab_octave/masses_chain_model/example_ocp.m [hunk 1]
```diff
@@ -158,9 +158,10 @@
 else
 	ocp_model.set('dyn_type', 'discrete');
 	ocp_model.set('dyn_expr_phi', model.expr_phi);
-	if (np>0)
-		ocp_model.set('dyn_param_phi', 'true');
-	end
+	% if (np>0)
```
Comment (asparc) [inline examples/acados_matlab_octave/masses_chain_model/example_ocp.m:161]: You can ignore this

File: interfaces/acados_template/acados_template/c_templates_tera/Makefile.in [hunk 2]
```diff
@@ -105,6 +105,25 @@
 	{%- set model_external_shared_lib_name = solver_options.model_external_shared_lib_name %}
 {%- endif %}
 
+{# control operator #}
```
[the remark this answers is not in the document]
Comment (FreyJo) [reply interfaces/acados_template/acados_template/c_templates_tera/Makefile.in:108]: basically applied all the changes from this file, but:
- appended openmp_flag to ACADOS_FLAGS
- didn't remove the includes

File: interfaces/acados_template/acados_template/c_templates_tera/make_sfun.in.m [hunk 3]
```diff
@@ -31,7 +31,7 @@
 % POSSIBILITY OF SUCH DAMAGE.;
 %
```
[the remark this answers is not in the document]
Comment (FreyJo) [reply interfaces/acados_template/acados_template/c_templates_tera/make_sfun.in.m:33]: Applied all the changes in this file. I had to remove the spaces in the `SOURCES` cell.

File: interfaces/acados_matlab_octave/ocp_solve.c [hunk 4]
```diff
@@ -40,10 +40,19 @@
...
 #include "mex.h"
 
-
+#if defined(_WIN32) || defined(__WIN32__) || defined(WIN32) || defined(_WIN64)
+#include <windows.h>
+#endif
 
 void mexFunction(int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[])
 {
+    // #if defined(_WIN32) || defined(__WIN32__) || defined(WIN32) || defined(_WIN64)
```
[the remark this answers is not in the document]
Comment (FreyJo) [reply interfaces/acados_matlab_octave/ocp_solve.c:49]: I applied the uncommented version of this change, as it should not break anything.

**Review by FreyJo — COMMENTED** [review]
Hi @asparc,
as discussed, instead of merging this branch, I applied the changes a bit more carefully.
I mainly had to do some changes for Octave compatibility, my Matlab version and the automatic MEX interface compilation.
I made this branch which contains very similar changes:
https://github.com/FreyJo/acados/tree/dev_windows_caspar
I will now test the changes on that branch on a Windows system.
If you have time, maybe try to do the same, and let me know if you find any issues.
The review here is thus just a preliminary feedback.
Anyhow, thanks a lot for the contribution already! :pray:

**Pull request closed without merging**